examples/data_collect/main.cpp

```diff
@@ -508,40 +499,48 @@ int main(int argc, char *argv[]) {
             }
 
             frame_size = sizeof(float) * height * width;
-            frameType = "conf";
-        }
-        // Raw data
-        else if (frame_type == "raw") {
-            aditof::FrameDataDetails rawFrameDetails;
-            frame.getDataDetails("raw", rawFrameDetails);
-
-            frame_size = rawFrameDetails.width * rawFrameDetails.height *
-                         rawFrameDetails.subelementsPerElement *
-                         rawFrameDetails.subelementSize;
-            frameType = "raw";
+        } else if (frame_type == "full-frame") {
+            FrameDataDetails frameDetails;
+            frame.getDataDetails("frameData", frameDetails);
+            frame_size =
+                sizeof(uint16_t) * frameDetails.height * frameDetails.width;
         } else {
             LOG(WARNING) << "Can't recognize frame data type!";
         }
 
-        /* Since getData returns the pointer to the depth/raw data, which only main thread has access to,
+        /* Since getData returns the pointer to the depth data, which only main thread has access to,
             we need to copy depth frame to local memory and pass that to thread for file I/O, there is no drop in the throughput with this. */
         frameBuffer = new uint16_t[frame_size];
         if (frameBuffer == NULL) {
             LOG(ERROR) << "Can't allocate Memory for frame type data!";
             return 0;
         }
 
-        uint16_t *pData;
-        status = frame.getData(frameType, &pData);
-        if (status != Status::OK) {
-            LOG(ERROR) << "Could not get frame type data!";
-            return 0;
-        }
-        if (!pData) {
-            LOG(ERROR) << "no memory allocated in frame";
-            return 0;
+        if (frame_type == "full-frame") {
```

Comment:
You could change frame_type value from "full-frame" to "frameData" somewhere at the beginning of data-collect. This will avoid many 'if' statements.

Reply:
Solved